Shared/AppInfo.swift

```diff
@@ -5,20 +5,34 @@
 import Foundation
 
 open class AppInfo {
-    open static var displayName: String {
-        return Bundle.main.object(forInfoDictionaryKey: "CFBundleDisplayName") as! String
+    /// Return the main application bundle. If this is called from an extension, the containing app bundle is returned.
+    open static var applicationBundle: Bundle? {
+        let bundle = Bundle.main
+        switch bundle.bundleURL.pathExtension {
+        case "app":
+            return bundle
+        case "appex":
+            // .../Client.app/PlugIns/SendTo.appex
+            return Bundle(url: bundle.bundleURL.deletingLastPathComponent().deletingLastPathComponent())
+        default:
+            return nil
+        }
+    }
+
+    open static var displayName: String? {
+        return applicationBundle?.object(forInfoDictionaryKey: "CFBundleDisplayName") as? String
     }
 
-    open static var appVersion: String {
-        return Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as! String
+    open static var appVersion: String? {
+        return applicationBundle?.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String
```

Comment:
I wonder if we should provide the default "0.0" value here instead of having to specify it as the fallback everywhere else?